esp_delta_ota/examples/https_delta_ota/tools/esp_delta_ota_patch_gen.py

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env python
+#
+# ESP Delta OTA Ptach Generator Tool. This tool helps in generating the compressed patch file
+# using BSDiff and Heatshrink algorithms
+#
+# SPDX-FileCopyrightText: 2023 Espressif Systems (Shanghai) CO LTD
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+import os
+import subprocess
+import re
+
+# Magic Byte is created using command: echo -n "esp_delta_ota" | sha256sum
+esp_delta_ota_magic = 0xfccdde10
+
+MAGIC_SIZE = 4
```

Comment:
We should mention this format in the top level README

Reply:
Added an image format in README